swf/src/read.rs

```diff
@@ -261,6 +261,18 @@ impl<'a, 'b> BitReader<'a, 'b> {
         self.bits.read_bit()
     }
 
+    /// A variant of `read_ubits` for when the number of bits to read is
+    /// a *c*ompile-*t*ime constant. This should be more efficient.
+    #[inline]
+    fn read_ubits_ct<const NUM_BITS: u32>(&mut self) -> io::Result<u32> {
+        if NUM_BITS > 0 {
+            self.bits.read::<NUM_BITS, u32>()
+        } else {
+            Ok(0)
+        }
+    }
+
+    /// If `num_bits` is a compile-time constant, consider `read_ubits_ct` instead.
```

Comment:
I recognize that strictly speaking this is not what a docstring is supposed to say, but... :no_mouth: ... this one is ... just ... like this! :shrug: